src/main/java/uk/gov/gchq/palisade/service/ResourceService.java

```diff
@@ -57,55 +52,54 @@
      * large collections of resources by requesting all the resources of one particular type. For example, a client may
      * request all "employee contact card" records. Please note the warning in the class documentation above, that just
      * because a resource is available does not guarantee that the requesting client has the right to access it.
      *
-     * @param request request object detailing the type of resource to retrieve.
-     * @return {@link CompletableFuture} that upon completion will contain the connection details for all resources
-     * matching a type
+     * @param type the String value of the resourceType being requested.
+     * @return a {@link Map} of the {@link LeafResource} and {@link ConnectionDetail} of how to retrieve the available
+     * resources matching the type
      */
-    CompletableFuture<Map<LeafResource, ConnectionDetail>> getResourcesByType(final GetResourcesByTypeRequest request);
+    Map<LeafResource, ConnectionDetail> getResourcesByType(final String type);
 
     /**
      * Find all resources that match a particular data format. Resources of a particular data format may not share a
      * type, e.g. not all CSV format records will contain employee contact details. This method allows clients to
      * retrieve all the resources Palisade knows about that conform to one particular format. Note that this method can
      * potentially return large ${@code Map}s with many mappings.
      *
-     * @param request the request detailing the specific format for retrieval
-     * @return a {@link CompletableFuture} that upon completion will contain the details on how to retrieve the
-     * resources
+     * @param format the String value of the resourceSerialisedFormat being requested.
+     * @return a {@link Map} of the {@link LeafResource} and {@link ConnectionDetail} of how to retrieve the available
+     * resources mtaching the format
      */
-    CompletableFuture<Map<LeafResource, ConnectionDetail>> getResourcesBySerialisedFormat(final GetResourcesBySerialisedFormatRequest request);
+    Map<LeafResource, ConnectionDetail> getResourcesBySerialisedFormat(final String format);
 
     /**
      * Informs Palisade about a specific resource that it may return to users. This lets Palisade clients request access
      * to that resource and allows Palisade to provide policy controlled access to it via the other methods in this
      * interface.
      *
-     * @param request details of the resource that Palisade can manage access to
-     * @return a {@link CompletableFuture} that will complete as true once the resource has been added to this {@link
-     * ResourceService}
+     * @param resource details of the resource that Palisade can manage access to.
+     * @return the {@link Resource} object that has been added to the {@link ResourceService}
      */
-    CompletableFuture<Boolean> addResource(final AddResourceRequest request);
+    Resource addResource(final Resource resource);
```

Comment:
While at the moment the Request object is only really delivering the Resource in terms of useful info for the resource service to use here. In future we may need to pass more information in such as the connection details as well, which is where having a request class we can easily extend makes it more extensible to future needs, without breaking compatibility to older versions.